src/components/pagination/bl-pagination.ts

```diff
@@ -80,10 +61,27 @@ export default class BlPagination extends LitElement {
   selectLabel = 'Show';
 
   /**
-   *  Sets the option texts.
+   *  Sets the items per page options of the select element
    */
-  @property({ attribute: 'option-text', type: String })
-  optionText = 'Items';
+  @property({ attribute: 'select-options', type: Array })
```

Comment:
I don't think it's a good idea to get serialized objects via attributes. I would prefer to get this via a property only (so it can be only set by JavaScript) or with a slot.

Passing stringified json as a property value can cause issues on FW implementations as well.

Can we reconsider this? Let's also see what others think about this.

Reply:
Another point here is the property/attribute name. `select-options` doesn't give correct feeling about what we are setting here. `select` is a representational choice for choosing an "item per page" value but, it would be a different element. To keep that decision internal to the component, I would suggest to use a name directly related with the content. Like `itemsPerPageOptions` maybe.

Reply:
I'm converting the pr to craft until we decide then. This is the first time we take an array property, so I was in the middle but now I think what you suggest makes total sense @muratcorlu . 
@leventozen  what do you think?

Reply:
I agree with @muratcorlu about his concerns.